tests/sweep_framework/sweeps/ccl/all_gather_n300.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case_n300
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+
+batch_sizes = [1, 2]
+heights = [32, 64, 128]
+widths = [32 * (2**i) for i in range(10)]  # [32, 64, 128, 256, 512, 1024, 2048, 4096, 8192, 16384]
+input_shapes = []
+for W in batch_sizes:
+    for Z in batch_sizes:
+        for height in heights:
+            for width in widths:
+                input_shapes.append([W, Z, height, width])
+
+parameters = {
+    "all_gather_n300": {
+        "num_devices": [2],
+        "num_links": [1],
+        "input_shape": input_shapes,
+        "dim": [0, 1, 2, 3],
+        "layout": [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT],
+        "input_dtype": [ttnn.bfloat16, ttnn.bfloat8_b],
+        "mem_config": [
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.DRAM),
+            ttnn.MemoryConfig(buffer_type=ttnn.BufferType.L1),
+        ],
+        "enable_async": [True],
+        "num_iters": [1],
+    },
+}
+
+
+def invalidate_vector(test_vector) -> Tuple[bool, Optional[str]]:
+    (is_known_failure, message) = is_unsupported_case_n300(
+        test_vector["input_shape"],
+        test_vector["dim"],
+        test_vector["mem_config"],
+        test_vector["num_devices"],
+        test_vector["num_links"],
+        test_vector["input_dtype"],
+        test_vector["layout"],
+    )
+    if is_known_failure:
+        return True, f"Skipping unsupported case {message}."
+    return False, None
+
+
+def mesh_device_fixture():
+    assert ttnn.get_num_devices() == 2, "Not N300!"
+
+    num_devices = ttnn.GetNumAvailableDevices()
+    device_ids = [i for i in range(num_devices)]
+
+    devices = ttnn.CreateDevices(device_ids)
+
+    yield ([devices[i] for i in range(num_devices)], "N300 Fixture")
+
```

Comment:
Is `del devices` going to properly close the devices? I think you should loop through and call `ttnn.close_device()` on each here

Reply:
@jdesousa-TT , i have tried looping through devices and closing but it seems like `ttnn.close_device(devices[0])` itself closes both 1 and 2 chip 
<img width="1163" alt="Screenshot 2024-09-27 at 7 06 38 PM" src="https://github.com/user-attachments/assets/1242131b-c3ee-43ad-88b8-df4723d4e679">

Reply:
That's fine, as long as they get closed properly. Thanks